crates/bevy_sprite/src/render/sprite.wgsl

```diff
@@ -2,33 +2,48 @@
 #import bevy_core_pipeline::tonemapping
 #endif
 
+#import bevy_render::maths affine_to_square
 #import bevy_render::view  View
 
 @group(0) @binding(0)
 var<uniform> view: View;
 
+struct VertexInput {
+    @builtin(vertex_index) index: u32,
+    // NOTE: Instance-rate vertex buffer members prefixed with i_
+    // NOTE: i_model_transpose_colN are the 3 columns of a 3x4 matrix that is the transpose of the
+    // affine 4x3 model matrix.
+    @location(0) i_model_transpose_col0: vec4<f32>,
+    @location(1) i_model_transpose_col1: vec4<f32>,
+    @location(2) i_model_transpose_col2: vec4<f32>,
+    @location(3) i_color: vec4<f32>,
+    @location(4) i_uv_offset_scale: vec4<f32>,
+}
+
 struct VertexOutput {
+    @builtin(position) clip_position: vec4<f32>,
     @location(0) uv: vec2<f32>,
-#ifdef COLORED
-    @location(1) color: vec4<f32>,
-#endif
-    @builtin(position) position: vec4<f32>,
+    @location(1) @interpolate(flat) color: vec4<f32>,
```

Comment:
why `@interpolate(flat)` here?

if for perf, it should probably be on uvs as well (unless we think people might use non-orthographic projections with a camera2d). although i just tried adding it to uvs and it didn't work, so i have misunderstood something.

Reply:
The colour is currently a per-instance variable, so it does not change between vertices and as such it does not need to be interpolated between vertices, which should give better performance. UVs are different at each vertex and so do need interpolating.